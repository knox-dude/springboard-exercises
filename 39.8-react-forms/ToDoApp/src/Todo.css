.Todo {
    display: flex;
    width: 200px;
    height: 50px;
    margin: 10px;
    border: 1px solid #ccc;
    justify-content: space-around;
}

.Todo-button {
    width: 50px;
    height: 50px;
    border: 1px solid #ccc;
    border-radius: 50%;
    background-color: #fff;
    cursor: pointer;
}