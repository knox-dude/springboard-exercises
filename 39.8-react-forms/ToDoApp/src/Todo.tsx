interface TodoProps {
  id: string;
  text: string;
  remove: (id: string) => void;
}

function Todo({ id, text, remove }: TodoProps) {

  const removeTodo = () => {
    remove(id);
  }

  return (
    <div>
      <p>{text}</p>
      <button onClick={removeTodo}>Remove</button>
    </div>
  );
}

export default Todo;
