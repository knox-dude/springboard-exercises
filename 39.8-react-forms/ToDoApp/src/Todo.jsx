import PropTypes from "prop-types";
import "./Todo.css"

function Todo({ task, id, remove }) {

  const handleDelete = () => {remove(id);};
  
  return (
    <div className="Todo">
      <p>{task}</p>
      <button className="Todo-button" onClick={handleDelete}>X</button>
    </div>
  );
}

Todo.propTypes = {
  task: PropTypes.string.isRequired,
  id: PropTypes.string.isRequired,
  remove: PropTypes.func,
};

export default Todo;
