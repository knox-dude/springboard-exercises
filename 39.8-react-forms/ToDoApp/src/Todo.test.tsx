import { render, screen, fireEvent } from '@testing-library/react';
import Todo from './Todo'; 

describe('Todo component', () => {
  const mockRemove = jest.fn(); // Mock the remove function

  //smoke test
  it('renders without crashing', () => {
    render(<Todo id="1" text="test" remove={mockRemove} />);
  });

  //snapshot test
  it('matches snapshot', () => {
    const { asFragment } = render(<Todo id="1" text="test" remove={mockRemove} />);
    expect(asFragment()).toMatchSnapshot();
  });

  it('calls remove function', () => {
    render(<Todo id="1" text="test" remove={mockRemove} />);
    const removeButton = screen.getByText('Remove');
    fireEvent.click(removeButton);
    expect(mockRemove).toHaveBeenCalled();
  });
});