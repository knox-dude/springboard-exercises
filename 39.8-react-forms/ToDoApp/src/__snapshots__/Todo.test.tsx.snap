// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Todo component matches snapshot 1`] = `
<DocumentFragment>
  <div>
    <p>
      test
    </p>
    <button>
      Remove
    </button>
  </div>
</DocumentFragment>
`;
